File: Data Structures & Algorithms/majority-element-ii/submission-0.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        freq = {}

        for n in nums:
            if n not in freq:
                freq[n] = 0
            freq[n] += 1

        res = []
        
        for n in freq.keys():
            if freq[n] > len(nums) // 3:
                res.append(n)
            
        return res

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
#include <vector>
#include <algorithm>
#include <unordered_map>
#include <iostream>

using namespace std;

class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> m;

        for(const int num: nums){
            m[num]++;
        }

        vector<pair<int, int>> sorted;

        for(const auto& pair: m){
            sorted.push_back({pair.second, pair.first});
        }

        sort(sorted.rbegin(), sorted.rend());

        vector<int> res;

        for(int i = 0; i < k; i++){
            res.push_back(sorted[i].second);
        }

        return res;

    }
};